src/client.ts

```diff
@@ -223,17 +228,24 @@ function fillInPathParams(params: { [key: string]: string }, path: string) {
 
 function makeMethod(adapter: ClientAdapter, handler: server.Handler, pathParams: string[]) {
   const params = paramObject(pathParams, handler.path);
-  const call = server.safe(handler.params, handler.query, handler.body, handler.response, ctx =>
-    adapter({ ...ctx, path: fillInPathParams(params, handler.path), servers: handler.servers })
+  const call = server.safe(
+    handler.headers,
+    handler.params,
+    handler.query,
+    handler.body,
+    handler.response,
+    ctx =>
+      adapter({ ...ctx, path: fillInPathParams(params, handler.path), servers: handler.servers })
   );
-  return (ctx: ClientArg<any, any>) =>
+  return (ctx: ClientArg<any, any, any>) =>
     call({
       path: handler.path,
       servers: handler.servers,
       method: handler.method,
       params,
-      query: (ctx as { query?: any }).query,
-      body: (ctx as { body?: any }).body
+      headers: ((ctx || {}) as { headers?: any }).headers,
```

Comment:
~maybe just being annoying, but I think `ctx ? ctx.headers : undefined` makes this easier for other people to read/understand.~ Actually, why not use your lib, `safe`, here instead?